controlplane/kubeadm/internal/control_plane.go

```diff
@@ -124,6 +124,17 @@ func (c *ControlPlane) MachineInFailureDomainWithMostMachines(machines Filterabl
 	return machineToMark, nil
 }
 
+// MachineWithDeleteAnnotation returns the machine that has delete annotatation key on it.
+func (c *ControlPlane) MachineWithDeleteAnnotation(machines FilterableMachineCollection) (*clusterv1.Machine, error) {
+	annotatedmachines := machines.Filter(machinefilters.HasAnnotationKey(controlplanev1.DeleteControlPlaneMachineAnnotation))
+	machineToPick := annotatedmachines.Oldest()
+	if machineToPick == nil {
+		return nil, errors.New("failed to pick control plane Machine with delete annotation key")
```

Comment:
I think this function doesn't need to return an error. It can be the case that no machine is marked by annotation, and that is normal.

Reply:
I think it is good to return error only if none of the machine is marked with annotation.